ros/api/common/utils.py

```diff
@@ -44,23 +49,22 @@ def check_for_user():
         if username is None:
             abort(403, message="Username doesn't exist")
 
-        return username
+        return ident, username
 
-    def check_for_system(inventory_id):
+    def check_for_system(ident, inventory_id):
         if not is_valid_uuid(inventory_id):
             abort(404, message='Invalid inventory_id, Id should be in form of UUID4')
 
-        system = System.query.filter(
-            System.inventory_id == inventory_id
-        ).first()
+        system_id = system_ids_by_org_id(ident['org_id']).filter(
```

Comment:
IMO, we can rate a recommendation for a system only when its performance profile is available. IIRC, we can see the system in ROS-UI when it has a performance profile. Is it good to check [systems_ids_for_existing_profiles](https://github.com/RedHatInsights/ros-backend/blob/main/ros/lib/utils.py#L166)?

Reply:
@r14chandra and @upadhyeammit  - On web UI, we also give option to rate for system when it is optimized with no suggestions. If you think we should allow this action ONLY when there will be suggestion(s) then as @upadhyeammit mentioned, could you please raise ticket and discuss with team? 
Thoughts?

Reply:
Are you just considering the `optimized` state?
My concern is about rating a system (which is not there in UI) via post request from API not from UI.
And as you mentioned about this state, we can consider handling separately as it would also need the UI fix as well.

Reply:
> Are you just considering the optimized state?

@r14chandra - Where? In here, I am not considering it. I mentioned that current web ui allowing to add rating even when there is no recommendations i.e specific to when system is optimized.

Reply:
> My concern is about rating a system (which is not there in UI) via post request from API not from UI.
This means we need to check for the performance profile table and not for the system table. 
Let me know if you didn't get it.

Reply:
@r14chandra  +1 for new change request. 
With this check, there will be inconsistency between web UI & API. 
As @upadhyeammit mentioned please create new change request, in my opinion we need to revise this feature completely.

Reply:
Created a card -> [RHIROS-1012](https://issues.redhat.com/browse/RHIROS-1012)

Reply:
Thanks!